File: src/simulation/agent/brainPackage/Behavior.ts
import { PolarCoordinate } from '../../common/PolarCoordinate';
import { Node } from '../../common/Node';
import { Speed } from '../../common/Speed';
import { State } from '../../common/State';

export abstract class Behavior {
  protected newPositions: PolarCoordinate[] = [];
  protected newNodes: Node[] = [];

  constructor() {}

  getNextState(newPositions: PolarCoordinate[], newNodes: Node[]): State {
    this.newPositions = newPositions;
    this.newNodes = newNodes;
    const theta = this.generateAngle();
    const speed = this.generateSpeed();
    return new State(theta, speed);
  }

  abstract generateAngle(): number;
  abstract generateSpeed(): Speed;
}

File: java/simulation/agent/brainPackage/Behavior.java
package simulation.agent.brainPackage;

import java.util.List;

import simulation.common.Node;
import simulation.common.PolarCoordinate;
import simulation.common.Speed;
import simulation.common.State;

public abstract class Behavior {

protected List<PolarCoordinate> newPostions;
protected List<Node> newNodes;

public Behavior(){
}

public State getNextState(List<PolarCoordinate> newPostions, List<Node> newNodes) {
	//System.out.println("get next state called!");
	this.newPostions = newPostions;
	this.newNodes = newNodes;
	int theta = generateAngle();
	Speed speed = generateSpeed();
	
	return new State(theta, speed);
	}

public abstract int generateAngle();

public abstract Speed generateSpeed();
}

